libminifi/src/core/yaml/YamlConfiguration.cpp

```diff
@@ -762,106 +762,109 @@ void YamlConfiguration::parsePortYaml(YAML::Node *portNode, core::ProcessGroup *
+    }
+    logger_->log_error("Invalid conversion for field %s. Value %s", propertyFromProcessor.getName(), propertyValueNode.as<std::string>());
+    if (exit_on_failure) {
+      // We do not exit here even if exit_on_failure is set. Maybe we should?
+      std::cerr << "Invalid conversion for " << propertyFromProcessor.getName() << " to " << defaultType.name() << std::endl;
+    }
+  }
+  return defaultValue;
+}
+
+void YamlConfiguration::parseSingleProperty(const std::string& propertyName, const YAML::Node& propertyValueNode, std::shared_ptr<core::ConfigurableComponent> processor) {
+  core::Property myProp(propertyName, "", "");
+  processor->getProperty(propertyName, myProp);
+  PropertyValue coercedValue;
+
+  // coerce the types. upon failure we will either exit or use the default value.
+  // we do this here ( in addition to the PropertyValue class ) to get the earliest
+  // possible YAML failure.
+  coercedValue = getValidatedProcessorPropertyForDefaultTypeInfo(myProp, propertyValueNode);
+
+  const std::string rawValueString = propertyValueNode.as<std::string>();
+  if (!processor->setProperty(myProp, coercedValue)) {
+    std::shared_ptr<core::Connectable> proc = std::dynamic_pointer_cast<core::Connectable>(processor);
+    if (proc != 0) {
+      logger_->log_warn("Received property %s with value %s but is not one of the properties for %s. Attempting to add as dynamic property.", propertyName, rawValueString, proc->getName());
+      if (!processor->setDynamicProperty(propertyName, rawValueString)) {
+        logger_->log_warn("Unable to set the dynamic property %s with value %s", propertyName.c_str(), rawValueString.c_str());
+      } else {
+        logger_->log_warn("Dynamic property %s with value %s set", propertyName.c_str(), rawValueString.c_str());
+      }
+    }
+  } else {
+    logger_->log_debug("Property %s with value %s set", propertyName.c_str(), rawValueString.c_str());
+  }
+}
+
+void YamlConfiguration::parsePropertyNodeElement(const std::string& propertyName, const YAML::Node& propertyValueNode, std::shared_ptr<core::ConfigurableComponent> processor,
+    const std::string &yaml_section) {
+  logger_->log_trace("Encountered %s", propertyName);
+  if (propertyValueNode.IsNull() || !propertyValueNode.IsDefined()) {
+    return;
+  }
+  if (propertyValueNode.IsSequence()) {
+    parsePropertyValueSequence(propertyName, propertyValueNode, processor, yaml_section);
+  } else {
+    parseSingleProperty(propertyName, propertyValueNode, processor);
+  }
+}
+
+void YamlConfiguration::parsePropertiesNodeYaml(YAML::Node *propertiesNode, std::shared_ptr<core::ConfigurableComponent> processor, const std::string &component_name,
+    const std::string &yaml_section) {
+  // Treat generically as a YAML node so we can perform inspection on entries to ensure they are populated
+  logger_->log_trace("Entered %s", component_name);
+  for (const auto propertyElem : *propertiesNode) {
```

Comment:
```suggestion
  for (const auto& propertyElem : *propertiesNode) {
```

Reply:
I think iterator value on a yaml node creates a custom type either way. This way this detail is not hidden:
https://github.com/apache/nifi-minifi-cpp/blob/main/thirdparty/yaml-cpp-yaml-cpp-20171024/include/yaml-cpp/node/detail/node_iterator.h#L144

Reply:
In this case I'm fine with either.